diesel/src/associations/mod.rs

```diff
@@ -1,18 +1,24 @@
 //! Traits related to relationships between multiple tables.
 //!
-//! **Note: This feature is under active development, and we are seeking feedback on the APIs that
-//! have been released. Please feel free to [open issues][open-issue], or join [our chat][gitter]
-//! to provide feedback.**
-//!
-//! [open-issue]: https://github.com/diesel-rs/diesel/issues/new
-//! [gitter]: https://gitter.im/diesel-rs/diesel
-//!
 //! Note: The derives in this guide are provided by `diesel_codegen`. Make sure you have
 //! `#[macro_use] extern crate diesel_codegen;` at the root of your crate.
 //!
-//! Associations in Diesel are bidirectional, but primarily focus on the child-to-parent
-//! relationship. You can declare an association between two records with
-//! `#[belongs_to]`.
+//! Associations in Diesel are unidirectional and focus on the child-to-parent
+//! relationship.
+//!
+//! **Child** refers to the *many* part of a *one to many* relationship and has *one parent*.
+//! **Parent** refers to the *one* part of a *one to many* relationship and can *have many
+//! children*.
+//! In the following relationship for our code examples, User has many Posts,
+//! so User is the parent and Posts are children.
+//!
+//! In order to declare this relationship, you must annotate your child struct
+//! with `#[derive(Associations)]` and `#[belongs_to(ParentStructName)]`.
+//! `#[belongs_to]` is given the name of the struct that represents the parent. Both parent and
+//! child types must implement the [`Identifiable`][identifiable] trait.
+//! The struct or table referenced in your association has to be in scope,
+//! so you'll need `use schema::posts` or similar to bring the
+//! table into scope.
 //!
 //! ```rust
 //! // You need to have the table definitions from `infer_schema!` or `table!` in scope to
```

Comment:
Just add those table definitions here?

Reply:
@weiznich Add the definitions for the doc-test example or re-word the preceding paragraph? The tables are imported via `doctest_setup.rs`.

Reply:
I think adding adding the actual definitions here will make the example easier to understand because the all needed code is included in the actual example. (This is not needed everywhere, but because associations are quite complex I think the docs may be improved by adding them here.)